test/modifiers.c

```diff
@@ -151,10 +157,67 @@ test_modmap_none(void)
     xkb_context_unref(context);
 }
 
+static void
+test_real_mods(void)
+{
+    struct xkb_context *context = test_get_context(0);
+    struct xkb_keymap *keymap;
+    const struct xkb_key *key;
+    xkb_keycode_t keycode;
+
+    keymap = test_compile_file(context, "keymaps/real-modifiers.xkb");
+    assert(keymap);
+
+    keycode = xkb_keymap_key_by_name(keymap, "AD01");
+    assert(keycode != XKB_KEYCODE_INVALID);
+    key = XkbKey(keymap, keycode);
+    assert(key->modmap == Mod5Mask);
+    assert(key->vmodmap == Mod10Mask << 1);
```

Comment:
to make sure I understand this right: `Mod10Mask` is the highest real one we have, and this key is mapped to the first vmod defined and thus must have `highest-real-mask << 1` set? Might be worth adding a comment to that effect, took me too long to work that out :)

Reply:
Good point! Fixed